near/devgigsboard
# Call socialdb to notify

This PR implements the contract part of Dev Gigs Board notification feature. It is currently deployed on `devgigsboard.bo.near`. The UI part is implemented and tested in https://near.social/#/devgigsboard.bo.near/widget/Ideas and https://near.social/#/devgigsboard.bo.near/widget/NotificationFeed.

# My test steps:
1. create two accounts, `alice.bo.near`, `bob.bo.near`, register in near.social, **go to the profile page, gives each of them a name**  (e.g. "Test Alice" / Test Bob") **!!!important**.
2. Sign in with `bob.bo.near`. Go to main board: https://near.social/#/devgigsboard.bo.near/widget/Ideas. The first time sign in to this fork's board, it will ask you to sign a transaction to allow `devgigsboard.bo.near` to set your `index/notify` key in social db. This is required because social db's permission model. Otherwise `devgigsboard.bo.near` cannot write state under your social db root key, and cannot trigger notification. It looks like this (screenshot is taken for `carl.bo.near`, bob's looks all same but arguments contains bob instead):
![image](https://user-images.githubusercontent.com/13259400/212234275-08b8d3a6-1743-4944-ada3-0dfc29e0bce7.png)

3. Submit an idea: https://near.social/#/devgigsboard.bo.near/widget/Ideas?postId=3
4. Sign out, sign in with Alice, like and comment to the idea.
5. Sign out, sign in with `devgigsboard.bo.near`, edit to the idea.
6. Sign out, sign in with Bob, observe there're 3 notifications. Go to https://near.social/#/devgigsboard.bo.near/widget/NotificationFeed you can see:

![image](https://user-images.githubusercontent.com/13259400/212236164-ab438be6-2299-4f68-8cbc-9682635db6e5.png)


Note that if you click to the notification bell, it will leads you to the original NotificationFeed widget instead of my fork, which doesn't recognize Dev Gigs type of notification:

![image](https://user-images.githubusercontent.com/13259400/212236215-e4d1e432-1376-4ecb-9658-8941afa86933.png)


I'll ask @evgenykuzyakov if he is okay to merge my fork.
